addons/repair/tests/test_repair.py

```diff
@@ -490,3 +490,86 @@ def test_repair_with_product_in_package(self):
         repair_order.action_repair_start()
         repair_order.action_repair_end()
         self.assertEqual(repair_order.state, 'done')
+
+    def test_repair_check_location(self):
+        # Test that the location of repair order has been updated correctly after the change lot.
+        self.product_a.tracking = 'lot'
+        self.product_a.type = 'product'
+
+        # Create two serial numbers
+        lot_A = self.env['stock.lot'].create({'name': 'lot_A', 'product_id': self.product_a.id})
+        lot_B = self.env['stock.lot'].create({'name': 'lot_B', 'product_id': self.product_a.id})
+        lot_C = self.env['stock.lot'].create({'name': 'lot_C', 'product_id': self.product_a.id})
+        lot_D = self.env['stock.lot'].create({'name': 'lot_D', 'product_id': self.product_a.id})
+
+        stock_location_shelf_1 = self.env['stock.location'].create({
+            'name': 'Shelf 1',
+            'location_id': self.stock_warehouse.lot_stock_id.id,
+        })
+        stock_location_shelf_2 = self.env['stock.location'].create({
+            'name': 'Shelf 2',
+            'location_id': self.stock_warehouse.lot_stock_id.id,
+        })
+
+        stock_location_Inventory_adjustment = self.env['stock.location'].create({
+            'name': 'Inventory adjustment',
+            'usage': 'inventory',
+        })
+
+        self.env['stock.quant']._update_available_quantity(self.product_a, self.stock_warehouse.lot_stock_id, 10, lot_id=lot_A)
+        self.env['stock.quant']._update_available_quantity(self.product_a, stock_location_shelf_1, 5, lot_id=lot_B)
+        self.env['stock.quant']._update_available_quantity(self.product_a, stock_location_shelf_2, 5, lot_id=lot_B)
+        self.env['stock.quant']._update_available_quantity(self.product_a, stock_location_Inventory_adjustment, 5, lot_id=lot_C)
+
+        # create a return picking
+        repair_form = Form(self.env['repair.order'])
+        repair_form.product_id = self.product_a
+        repair_form.lot_id = lot_A
+        self.assertEqual(repair_form.location_id, self.stock_warehouse.lot_stock_id)
```

Comment:
to make the test more coherent, we need to check that it does not go in the default location
```suggestion
        self.assertEqual(repair_form.location_id, stock_location_shelf_1)
```